src/esp32/src/flash.c

```diff
@@ -4,16 +4,148 @@
  * SPDX-License-Identifier: Apache-2.0 OR MIT
  */
 
-#include "flash.h"
+#include <target/flash.h>
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <log.h>
+#include <private/rom.h>
+#include <private/soc_utils.h>
+
+#include "soc/gpio_reg.h"
+#include "soc/spi_reg.h"
+
+#define SPI_FLASH_CMD_RDID 0x9F
+#define ESP_ROM_SPIFLASH_BUSY_FLAG BIT(0)
+
+#define PERIPHS_SPI_FLASH_CMD SPI_CMD_REG(1)
+#define PERIPHS_SPI_FLASH_STATUS SPI_RD_STATUS_REG(1)
+#define PERIPHS_SPI_FLASH_USRREG SPI_USER_REG(1)
+#define PERIPHS_SPI_FLASH_USRREG2 SPI_USER2_REG(1)
+#define PERIPHS_SPI_FLASH_C0 SPI_W0_REG(1)
+#define PERIPHS_SPI_MOSI_DLEN_REG SPI_MOSI_DLEN_REG(1)
+#define PERIPHS_SPI_MISO_DLEN_REG SPI_MISO_DLEN_REG(1)
+#define SPI_USR2_DLEN_SHIFT SPI_USR_COMMAND_BITLEN_S
+
+/* ROM */
+
+extern uint32_t ets_efuse_get_spiconfig(void);
+extern void esp_rom_spiflash_attach(uint32_t ishspi, bool legacy);
+extern esp_rom_spiflash_chip_t g_rom_flashchip;
+extern uint8_t g_rom_spiflash_dummy_len_plus[];
 
 void stub_target_flash_init(void *state)
 {
+    STUB_LOG_TRACE();
```

Comment:
Lets move `(void)variable;` statements at the first line of the functions. Then local variable definitions, trace logs.. function calls. .